impl.go

```diff
@@ -32,34 +34,62 @@ type Pool interface {
 }
 
 type poolImpl struct {
-	cl *gsrpc.SubstrateAPI
+	urls []string
+	cl   *gsrpc.SubstrateAPI
+	r    int
+	m    sync.Mutex
 }
 
-func NewPool(url string) (Pool, error) {
-	cl, err := gsrpc.NewSubstrateAPI(url)
-	if err != nil {
-		return nil, err
+func NewPool(url ...string) (Pool, error) {
+	if len(url) == 0 {
+		return nil, fmt.Errorf("at least one url is required")
 	}
 
 	return &poolImpl{
-		cl: cl,
+		urls: url,
+		r:    rand.Intn(len(url)), // start with random url, then roundrobin
```

Comment:
I think it's better to shuffle the list of urls instead so it is randomized, instead of starting at a random index. Like this, if one of the endpoints is down, all of the connections to that one will be moved to the next endpoint in the list, which will cause that one to have way more load than the others. If the list is just randomized here when it's created, this should not be a problem.